### ISSUE TITLE
Create explicit hash storage and retrieval methods in the new entity storage classes

### ISSUE THREAD

**Issue by ghost** [GetDKAN/dkan#3465]
## User story: 
As an admin, when using the new custom entity system I would like to see readable titles for all my metadata so that I can find what I'm looking for.

## Acceptance criteria:
- We can submit new data and the entities for the datasets are created.
- We can see readable titles for all created entities.
- Hashes are stored separately from the titles but referencer still works.

Blocked by #3321 
Merge again the feature branch. 

## Estimate: 
2 days

**Comment by dafeder** [comment]
I don't think we need a separate issue for this anymore, assuming we get back to the custom entity this is just a requirement
